DogScepterLib/Core/GMData.cs

```diff
@@ -4,109 +4,195 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
-using System.Text;
+using System.Linq;
 using System.Threading.Tasks.Dataflow;
 
-namespace DogScepterLib.Core
-{
-    public delegate void Log(string message);
+namespace DogScepterLib.Core;
+
+/// <summary>
+/// A delegate that is used for logging messages.
+/// </summary>
+public delegate void Log(string message);
 
-    public class GMData
+/// <summary>
+/// Class that represents a GameMaker data file.
+/// </summary>
+public class GMData
+{
+    /// <summary>
+    /// Class that includes version information about a GameMaker data file.
+    /// </summary>
+    public class GMVersionInfo
     {
-        public class GMVersionInfo
+        /// <summary>
+        /// The major version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Major = 1;
+
+        /// <summary>
+        /// The minor version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Minor = 0;
+
+        /// <summary>
+        /// The release version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Release = 0;
+
+        /// <summary>
+        /// The build version of the IDE a GameMaker data file was built with.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.</remarks>
+        public int Build = 0;
+
+        /// <summary>
+        /// Indicates the bytecode format of the GameMaker data file.
+        /// </summary>
+        /// <remarks>This is only an approximation, as some IDE versions do not increase this number.
+        /// For example, this has a value of <c>17</c> from version 2.2.1 since at least 2022.3. </remarks>
+        public byte FormatID = 0;
+
+        /// <summary>
+        /// Whether the GameMaker data file aligns chunks to 16 bytes.
+        /// </summary>
+        /// <remarks>GameMaker versions after Game Maker: Studio 2.2.2 use this. On older versions,
```

Comment:
Sorry for the very small nitpick, but it's going to bother me, so I just wanted to mention that the colon is omitted in GMS2.